go-controller/pkg/cluster/management-port.go

```diff
@@ -46,35 +46,42 @@ func createManagementPortGeneric(nodeName, localSubnet string, clusterSubnet []s
 		"external-ids:iface-id=k8s-"+nodeName)
 	if err != nil {
 		logrus.Errorf("Failed to add port to br-int, stdout: %q, stderr: %q, error: %v", stdout, stderr, err)
-		return "", "", "", "", err
+		return "", "", "", "", "", err
 	}
 	macAddress, err := util.GetOVSPortMACAddress(interfaceName)
 	if err != nil {
 		logrus.Errorf("Failed to get management port MAC address: %v", err)
-		return "", "", "", "", err
+		return "", "", "", "", "", err
 	}
 
-	// Create this node's management logical port on the node switch. Now that the second subnet IP is
-	// statically allocated to the management logical port, we can safely remove the other_config:exclude_ips
-	// in the same transaction to avoid "Duplicate IP set" warning messages in ovn-northd.
-	ip = util.NextIP(ip)
-	portIP := ip.String()
-	n, _ := subnet.Mask.Size()
-	portIPMask := fmt.Sprintf("%s/%d", portIP, n)
-	stdout, stderr, err = util.RunOVNNbctl("--", "--may-exist", "lsp-add", nodeName, "k8s-"+nodeName,
-		"--", "lsp-set-addresses", "k8s-"+nodeName, macAddress+" "+portIP,
-		"--", "--if-exists", "remove", "logical_switch", nodeName, "other-config", "exclude_ips")
+	// Create this node's management logical port on the node switch.
+	_, portIP, n, err := util.GetNodeWellKnownAddresses(subnet)
 	if err != nil {
 		logrus.Errorf("Failed to add logical port to switch, stdout: %q, stderr: %q, error: %v", stdout, stderr, err)
-		return "", "", "", "", err
+		return "", "", "", "", "", err
 	}
+	portIPMask := fmt.Sprintf("%s/%d", portIP, n)
+
 	// switch-to-router ports only have MAC address and nothing else.
 	routerMac, stderr, err := util.RunOVNNbctl("lsp-get-addresses", "stor-"+nodeName)
 	if err != nil {
 		logrus.Errorf("Failed to retrieve the MAC address of the logical port, stderr: %q, error: %v",
 			stderr, err)
-		return "", "", "", "", err
+		return "", "", "", "", "", err
 	}
 
-	return interfaceName, portIPMask, routerIP, routerMac, nil
+	return interfaceName, portIPMask, macAddress, routerIP, routerMac, nil
+}
+
+// ManagementPortReady will check to see if the portMac was created
+func ManagementPortReady(portName string) (bool, error) {
+	portMac, portIP, err := util.GetPortAddresses(portName)
```

Comment:
You cannot use `GetPortAddresses()` here. It is meant for logical switch port with addresses set to `dynamic` during the logical switch port creation.

In our case, the management logical switch port is created with static IP and MAC, so you need to check for "addresses" column in the logical_switch_port row.

Reply:
@girishmg I had a commit in my extensions branch that updated GetPortAddresses() to fall back to static addreses, which @mccv1r0 has cherry-picked locally and will push soon.

We will also need that patch for a PR that @alexanderConstantinescu is working on to fix handling of existing pods if the OVN database is empty on restart. That used to work, but got broken when we removed the old HA support.